### PR TITLE
Base64: Increase performance by not using string::find

### PR DESCRIPTION
**Pull request by appgurueu**
Title says it all. Instead of iterating all possible characters and comparing, it uses the ASCII ranges to compare & subtract.

### CODE REVIEW

**Comment by lhofhansl** [comment]
Clever. Did you see this as a problem in a profiler. Or just a "random" improvement? (Nothing wrong with that, just curious)

**Comment by appgurueu** [comment]
> Clever. Did you see this as a problem in a profiler. Or just a "random" improvement? (Nothing wrong with that, just curious)

Random improvement, I was having a look at the code for the validation fix and noticed that it was using string::find.

**Comment by numberZero** [comment]
You’re optimizing the wrong part of the function. Besides, there’s UB if the string length is not divisible by 4 as 0 (ASCII NUL, not digit zero) is passed to your function in that case.

**Comment by appgurueu** [comment]
> It seems you’re optimizing the wrong part of the function. Besides, there’s UB if the string length is not divisible by 4 as 0 (ASCII NUL, not digit zero) is passed to your function in that case.

Should be addressed now.

File: src/util/base64.cpp [hunk 1]
```diff
@@ -39,6 +39,7 @@ static inline bool is_base64(unsigned char c)
 	return isalnum(c) || c == '+' || c == '/' || c == '=';
 }
 
+static const size_t nullIndex = base64_chars.find((char)0);
```
Comment (nerzhul) [inline src/util/base64.cpp:42]: beltter that const, const constexpr to calculat it at compilation time

**Review by nerzhul — CHANGES_REQUESTED** [review]

**Comment by rubenwardy** [comment]
Closed as no response since action/change needed

**Pull request closed without merging**